### PR TITLE
migrate pallet-offences-benchmarking to benchmark v2 syntax

### PR DESCRIPTION
**Pull request by clangenb**
Migrates pallet-offences-benchmarking to benchmark v2 syntax.

Part of:
* #6202

### CODE REVIEW

File: substrate/frame/offences/benchmarking/src/inner.rs [hunk 1]
```diff
@@ -144,7 +144,7 @@ fn create_offender<T: Config>(n: u32, nominators: u32) -> Result<Offender<T>, &'
 fn make_offenders<T: Config>(
 	num_offenders: u32,
 	num_nominators: u32,
-) -> Result<(Vec<IdentificationTuple<T>>, Vec<Offender<T>>), &'static str> {
+) -> Result<Vec<IdentificationTuple<T>>, &'static str> {
```
Comment (clangenb) [inline substrate/frame/offences/benchmarking/src/inner.rs:147]: `Offenders` was unused, and it was a private function, so I removed it.

File: substrate/frame/offences/benchmarking/src/mock.rs [hunk 2]
```diff
@@ -66,7 +67,8 @@ sp_runtime::impl_opaque_keys! {
 
 pub struct TestSessionHandler;
 impl pallet_session::SessionHandler<AccountId> for TestSessionHandler {
-	const KEY_TYPE_IDS: &'static [sp_runtime::KeyTypeId] = &[];
+	// corresponds to the opaque key id above
+	const KEY_TYPE_IDS: &'static [KeyTypeId] = &[KeyTypeId([100u8, 117u8, 109u8, 121u8])];
```
Comment (clangenb) [inline substrate/frame/offences/benchmarking/src/mock.rs:71]: Needed to fix genesis integrity check error.

**Comment by clangenb** [comment]
@re-gius review bump :)

**Comment by re-gius** [comment]
bot bench substrate-pallet --features=runtime-benchmarks --pallet=pallet_offences_benchmarking

**Comment by re-gius** [comment]
bot bench substrate-pallet --features=runtime-benchmarks --pallet=pallet_offences

**Review by gui1117 — APPROVED** [review]

**Review by re-gius — APPROVED** [review]

**Pull request merged**